contracts/instruments/RibbonVolatility.sol

```diff
@@ -459,12 +459,48 @@ contract RibbonVolatility is DSMath, InstrumentStorageV1, InstrumentStorageV2 {
         emit Exercised(msg.sender, positionID, totalProfit, optionsExercised);
     }
 
-    function claimRewards(string calldata adapterName, address rewardsAddress, uint256[] calldata optionIDs)
+    function claimRewards(string calldata adapterName, address rewardsAddress)
+        external
+    {
+      IProtocolAdapter adapter = IProtocolAdapter(factory.getAdapter(adapterName));
+      uint256[] memory optionIDs = getOptionIDs(msg.sender);
+      uint256 claimedRewards = adapter.delegateClaimRewards(rewardsAddress, optionIDs);
+      emit ClaimedRewards(claimedRewards);
+    }
+
+    function rewardsClaimable(string calldata adapterName, address rewardsAddress)
         external
+        view
+        returns (uint256 rewardsToClaim)
     {
       IProtocolAdapter adapter = IProtocolAdapter(factory.getAdapter(adapterName));
```

Comment:
Can we just revert here if `adapterName != "HEGIC"`? It will close us off to some weird scenarios.

When Opyn launches their rewards we'll upgrade and change it anyway.